server/route.go

```diff
@@ -1255,7 +1253,7 @@ func (c *client) parseUnsubProto(arg []byte) (string, []byte, []byte, error) {
 		return _EMPTY_, nil, nil, fmt.Errorf("parse error: '%s'", arg)
 	}
 	if accountName == _EMPTY_ {
-		accountName = string(args[0])
+		accountName = bytesToString(args[0])
```

Comment:
No, that would reference the read buffer and the accountName would later be used as a key to store in some maps.

This is called by `processGatewayRUnsub()`, which then uses account for looking up account (see issue reported earlier with that) and would use it to store an object here: `c.gw.outsim.Store(accName, e)`.
This is also called by `processRemoteUnsub()` which possibly calls `gatewayUpdateSubInterest()` which again uses the account name as key to some map.